todo_app/login/authorization.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+from flask import abort, current_app
+from flask_login import current_user
+
+READER = 'READER'
+WRITER = 'WRITER'
+
+
+def writer_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if not current_user_can_write():
+            abort(403)

```

Comment:
One learner I saw recently chose to make this a 404 which I thought was really interesting as a way of saying attackers shouldn't be allowed to glean info about whether an endpoint exists if they don't have permission - it has a trade off depending on what you want to communicate to legitimate users though! I thought [this article discussed it well](https://www.bennadel.com/blog/2400-handling-forbidden-restful-requests-401-vs-403-vs-404.htm#:~:text=The%20three%20status%20codes%20that,404%20%2D%20Not%20Found)